app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,138 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+@SpringBootTest
+class TaskControllerTest {
+    private TaskController taskController;
+
+    private TaskService taskService = new TaskService();
+    private static final String DUMMY_TITLE = "dummy_1";
+
+    @BeforeEach
+    public void init(){
+        taskController = new TaskController(taskService);
+        Task task1 = new Task();
+        task1.setTitle(DUMMY_TITLE);
+
+        taskController.create(task1);
+    }
+
+    @DisplayName("/tasks Process")
+    @Nested
+    class ListTaskController{
+
+        @DisplayName("/tasks 테스트")
+        @Test
+        public void list() {
+            TaskService service = new TaskService();
+            TaskController controller = new TaskController(taskService);
+            assertThat(controller.list()).isNotEmpty();
+        }
+
+        @DisplayName("/tasks Process는 할일을 아이디로 조회해서 할일을 반환합니다.")
+        @Test
+        public void PathListValid(){
+            //when
+            assertThat(taskController.detail(1L).getTitle()).isEqualTo(DUMMY_TITLE);
+        }
+
+        @DisplayName("detailInValidId 오류 확인")
+        @Test
+        public void PathListInvalid(){
+            assertThatExceptionOfType(TaskNotFoundException.class)
+                    .isThrownBy(()->{
+                        taskController.detail(100L);
+                    }).withMessage("Task not found: 100");
+        }
+    }
+
+
+    @DisplayName("Create Task Process")
+    @Nested
+    class CreateProcess{
+
+        @DisplayName("valid Create Task")
+        @Test
+        public void createNewTask() {
+            assertThat(taskController.list()).isNotEmpty();
+        }
+
+        @DisplayName("Invalid Create Task")
+        @Test
+        public void InvalidCreateTask() {
+            assertThat(taskController.create(new Task()).getTitle()).isNull();
+        }
+
+
+    }
+
+    @DisplayName("update Process")
+    @Nested
+    class UpdateProcess{
+
+        @DisplayName("update getId")
+        @Test
+        public void update() {
+            //given
+            Task task1 = new Task();
+            task1.setTitle("new Task");
+            taskController.update(1L , task1);
+            //when
+            assertThat(taskController.list().get(0).getTitle()).isEqualTo("new Task");
+            //Then
+        }
+    }
+
+    @DisplayName("Delete Task Process")
+    @Nested
+    class DeleteTaskController{
+
+        @DisplayName("Delete Task 확인")
+        @Test
+        public void deleteController(){
+            //when
+            taskController.delete(1L);
+            //Then
+            assertThat(taskController.list()).isEmpty();
+        }
+    }
```

Comment:
만약에 이 테스트가 실패한다고 했을 때, 무엇이 잘못되었는지 이해하기 어렵습니다. `확인`이라는 말이 아무렇게나 해석될 수 있기 때문인데요. 명확하게 표현해주는게 좋을 것 같습니다. 예를들어서 저라면 이렇게 할 것 같습니다.

```suggestion
    @DisplayName("Delete")
    @Nested
    class DeleteTaskController{

        @DisplayName("할 일을 삭제합니다.")
        @Test
        public void deleteController(){
            //when
            taskController.delete(1L);
            
            //Then
            assertThat(taskController.list()).isEmpty();
        }
    }
```

그러면 테스트가 깨졌을 때 삭제되어야 하는데 삭제되지 않은 것이니까 더 파악하기 쉬울 것 같습니다